### PR TITLE
IPC Message clean up

### PR DESCRIPTION
**Pull request by gatesn**
Part of #1676

### CODE REVIEW

File: vortex-flatbuffers/flatbuffers/vortex-serde/message.fbs [hunk 1]
```diff
@@ -9,32 +9,10 @@ enum Compression: uint8 {
     None = 0,
 }
 
-table Schema {
```
Comment (AdamGS) [inline vortex-flatbuffers/flatbuffers/vortex-serde/message.fbs:12]: Do we want to completely give up the idea of a paruqet-like footer with arbitrary metadata/hints?

File: vortex-flatbuffers/src/lib.rs [hunk 2]
```diff
@@ -145,14 +146,19 @@ pub trait WriteFlatBuffer {
 }
 
 pub trait FlatBufferToBytes {
-    fn with_flatbuffer_bytes<R, Fn: FnOnce(&[u8]) -> R>(&self, f: Fn) -> R;
+    /// Write the flatbuffer to a byte vector and return the vector along with the starting
```
Comment (AdamGS) [inline vortex-flatbuffers/src/lib.rs]: nit - `vector` -> `buffer`

Comment (AdamGS) [reply vortex-flatbuffers/src/lib.rs]: \+ the name is confusing given that there's also a `WriteFlatBuffer` trait

File: vortex-flatbuffers/flatbuffers/vortex-serde/message.fbs (hunk 1 above)
Comment (gatesn) [reply vortex-flatbuffers/flatbuffers/vortex-serde/message.fbs:12]: I'm not sure what you mean? This is for the IPC format, seemed pointless wrapping DType again

**Review by AdamGS — APPROVED** [review]

**Pull request merged**